### PR TITLE
Linfa's 0.4.0 release

### PR DESCRIPTION
**Pull request by bytesnake**
These are the notes for the 0.4.0 release. If you want to add something to the release notes or have some objections please write them here. I'm planing to merge the PR in the next two days. @Sauro98 @YuhanLiin @relf @quietlychris.

### CODE REVIEW

**Review by Sauro98 — APPROVED** [review]
I like the notes, I just pointed out a typo 👍🏻 Also there's an empty "Example" subtitle

File: docs/website/content/news/release_040.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+date = "2021-04-27"
++++
+
+Linfa's 0.4.0 release introduced four new algorithms, improved documentation of ICA and K-means, added more benchmarks to K-Means and updated to ndarray's 0.14 version.
+
+<!-- more -->
+
+## New algorithms
+
+The [Partial Least Squares Regression](https://en.wikipedia.org/wiki/Partial_least_squares_regression) model family is added in this release. It projects the observable, as well as predicted variables to a latent space and maximizes the correlation in it. For problems with a large number of targets or collinear predictors it gives a better performance when compared to standard regression. For more information look into the documentation of `linfa-psa`.
```
[the remark this answers is not in the document]
Comment (bytesnake) [reply docs/website/content/news/release_040.md:12]: yes

**Comment by bytesnake** [comment]
> I like the notes, I just pointed out a typo 👍🏻 Also there's an empty "Example" subtitle

haven't finished it yesterday, I also want to add a figure of t-SNE

**Comment by Sauro98** [comment]
Sorry, I somehow did not see that this pr was marked as a draft, that's why I reviewed it 😅

**Comment by bytesnake** [comment]
@Sauro98 I also added an example for `linfa-preprocessing`. It's showing the use TF-IDF for text preprocessing and basically just copies the example and goes through it with some text.

**Comment by Sauro98** [comment]
Cool 👍🏻 
My only suggestion is to also add the line in which the training dataset is created as you do for the test dataset in [line 134](https://github.com/rust-ml/linfa/pull/127/files#diff-f1f6c13ee213e47059763397fb1096ba87603b042a22ae4e8d41bfbb5ab554ebR134), but that's just a preference, since it's not supposed to be a runnable example

**Comment by bytesnake** [comment]
> Cool 👍🏻
> My only suggestion is to also add the line in which the training dataset is created as you do for the test dataset in [line 134](https://github.com/rust-ml/linfa/pull/127/files#diff-f1f6c13ee213e47059763397fb1096ba87603b042a22ae4e8d41bfbb5ab554ebR134), but that's just a preference, since it's not supposed to be a runnable example

I've now just added a comment saying that I've omitted the target's and dataset's construction. Because the function call gives no additional information for the reader. Are we ready to publish?

**Comment by Sauro98** [comment]
> I've now just added a comment saying that I've omitted the target's and dataset's construction, because the function call gives no additional information for the reader.

Yes, that's probably better

> Are we ready to publish?

I have no other comments about the release notes/snippets. I noticed that in the base crate documentation, inside the table that lists all subcrates, I accidentally copied the sub-crate relative link for preprocessing instead of putting the docs.rs link. The same goes for pls and tsne

**Comment by bytesnake** [comment]
ups missed that, I've also added some comments on the new lapack trait bound in the `CONTRIBUTE.md`

**Review by relf — APPROVED** [review]
Looks good to me. `With/WithoutLapack` trait is a really nice mechanism to simplify the code, without mentioning the `F::cast()` helper! :clap: :clap: :clap:

**Pull request merged**